cpp/include/raft/cudart_utils.h

```diff
@@ -388,6 +388,12 @@ void ASSERT_DEVICE_MEM(T* ptr, std::string name)
               << ", err=" << s_err << std::endl;
 }
 
+#define __ASSERT_STR_HELPER(x) #x
+
+#define RAFT_DEVICE_ASSERT(cond) \
```

Comment:
Would it make sense to enable this only for the Debug build, like e.g. `RAFT_CHECK_CUDA`?

Reply:
I think we can use the `assert` macro for debugging purposes.  This function is specifically written to perform checks regardless of the build configuration. (Currently, I use it in test files only)

Reply:
Then, perhaps, you can move it from this public header to somewhere in tests?

Reply:
Thank you for the suggestion, I removed the assert function.